lib/debug/session.rb

```diff
@@ -1654,15 +1654,53 @@ def method_added tp
               end
             end
 
-            unresolved = true unless bp.enabled?
+            resolved = false if !bp.enabled?
           end
         }
-        unless unresolved
-          METHOD_ADDED_TRACKER.disable
+
+        if resolved
+          Session.method_added_tracekr_deactivate
+        end
+
+        case mid
+        when :method_added, :singleton_method_added
+          Session.method_added_tracekr_create(tp.self, mid)
+          Session.method_added_tracekr_activate unless resolved
         end
       end
     end
 
+    class ::Module
+      undef method_added
+      def method_added mid; end
+      def singleton_method_added mid; end
+    end
+
+    def self.method_added_tracekr_create mod, method_added_id, method_accessor = :method
+      m = mod.__send__(method_accessor, method_added_id)
+      METHOD_ADDED_TRACKERS[m] = TracePoint.new(:call) do |tp|
+        SESSION.method_added tp
+      end
+    end
+
+    def self.method_added_tracekr_activate
```

Comment:
`method_added_tracker_activate`?
I'd also recommend putting the verb `activate` in front of the method. So it'd be `activate_method_added_tracker`.